modules/core/src/main/java/org/apache/ignite/internal/commandline/checkpointing/CheckpointingForceTask.java

```diff
@@ -0,0 +1,136 @@
+import org.apache.ignite.compute.ComputeJobResult;
+import org.apache.ignite.internal.IgniteInternalFuture;
+import org.apache.ignite.internal.commandline.meta.subcommands.VoidDto;
+import org.apache.ignite.internal.processors.cache.persistence.CheckpointState;
+import org.apache.ignite.internal.processors.cache.persistence.checkpoint.CheckpointProgress;
+import org.apache.ignite.internal.processors.task.GridInternal;
+import org.apache.ignite.internal.util.future.GridFutureAdapter;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.internal.visor.VisorJob;
+import org.apache.ignite.internal.visor.VisorMultiNodeTask;
+import org.apache.ignite.lang.IgniteInClosure;
+import org.apache.ignite.resources.JobContextResource;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Task for checkpointing operation
+ */
+@GridInternal
+public class CheckpointingForceTask extends VisorMultiNodeTask<VoidDto, CheckpointingForceResult, NodeCheckpointingResult> {
+
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    @Override protected VisorJob<VoidDto, NodeCheckpointingResult> job(VoidDto arg) {
+        return new CheckpointingForceJob(debug);
+    }
+
+    @Nullable
+    @Override protected CheckpointingForceResult reduce0(List<ComputeJobResult> results) throws IgniteException {
+        CheckpointingForceResult res = new CheckpointingForceResult();
+
+        for (ComputeJobResult result : results) {
+            NodeCheckpointingResult nodeRes = result.getData();
+            res.status().put(nodeRes.nodeId(), nodeRes);
+        }
+        return res;
+    }
+
+    /** */
+    private static class CheckpointingForceJob extends VisorJob<VoidDto, NodeCheckpointingResult> {
+
+        /** */
+        private static final long serialVersionUID = 0L;
+
+        /** */
+        @JobContextResource
+        private ComputeJobContext jobCtx;
+
+        /** */
+        private GridFutureAdapter fut;
+
+        private NodeCheckpointingResult res;
+
+        long startTime;
+
+        /**
+         * @param arg Argument.
+         * @param debug Debug.
+         */
+        public CheckpointingForceJob(boolean debug) {
+            super(null, debug);
+        }
+
+        /** {@inheritDoc} */
+        @Override protected NodeCheckpointingResult run(@Nullable VoidDto arg) throws IgniteException {
+            if (fut == null) {
+                res = new NodeCheckpointingResult();
+                res.nodeId(ignite.localNode().id());
+
+                startTime = U.currentTimeMillis();
+                CheckpointProgress progress = ignite.context().cache().context().database().forceCheckpoint("force");
+                if (progress == null) {
+                    res.durationMillis(U.currentTimeMillis() - startTime);
+                    return res;
+                }
+
+                fut = progress.futureFor(CheckpointState.FINISHED);
+                if (fut.isDone())
```

Comment:
In case of the `fut` is already done here, this implementation always returns `null` instead of the correct result.